### ISSUE TITLE
grid-cols arbitrary value not working as expected

### ISSUE THREAD

**Issue by squelix** [tailwindlabs/tailwindcss#12281]
**What version of Tailwind CSS are you using?**
v3.3.4

**What build tool (or framework if it abstracts the build tool) are you using?**
Next.js 13.5.6

**What version of Node.js are you using?**
v16

For example: v16

**What browser are you using?**
Chrome

**What operating system are you using?**
macOS

**Describe your issue**

I have this arbitrary value in className
```
"grid grid-cols-[repeat(3,_minmax(0,_max-content))]"
```
It's not generating the good css. 

```
grid-template-columns: repeat(3,minmax(0,max - content));
```

There is spaces around the dash.

In tailwind v3.3.3 it was working, so I downgrade the version.

**Comment by DarkPromise** [comment]
Having the same issue here when using "grid-cols-[repeat(5,minmax(auto,max-content))]"
The new spaces injected causes the CSS to break and become invalid.

**Comment by JocelynLb** [comment]
I have the same issue with this line
` h-[calc(100%-env(safe-area-inset-bottom,0)-var(--main-size)-var(--bottom-size))] `
which is transformed into:
` height: calc(100% - env(safe - area - inset - bottom,0) - var(--main-size) - var(--bottom-size));.`

**Comment by RobinMalfait** [comment]
Hey, thank you for this bug report! 🙏

We introduced a regression here, but this should be fixed by #12283, and is available in the latest release ([3.3.5](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.3.5)). Sorry for the inconvenience!
